src/server/types.ts

```diff
@@ -33,20 +33,24 @@ export interface IslandModule {
 }
 
 export interface HandlerContext<T = unknown> {
+  connInfo: ConnInfo;
```

Comment:
I think `HandlerContext` should implement `ConnInfo` rather than it containing a `ConnInfo` object. That way you can do `ctx.sourceIp` rather than having to do `ctx.connInfo.sourceIp`.

Reply:
done